### ISSUE TITLE
plugin fixtures work only for a single test case

### ISSUE THREAD

**Issue by ltalirz** [aiidateam/aiida_core#1425]
To reproduce, use this `test_12.py`
```
from aiida.utils.fixtures import PluginTestCase

class TestCase1(PluginTestCase):
    """
    Test the PluginTestcase from utils.fixtures
    """
    BACKEND = 'django'

    def test_1(self):
        print("Running test 1")

class TestCase2(PluginTestCase):
    """
    Test the PluginTestcase from utils.fixtures
    """
    BACKEND = 'django'

    def test_2(self):
        print("Running test 2")
```

And run then run
```
$ python -m unittest discover
Running test 1
.E
======================================================================
ERROR: setUpClass (test_abc.TestCase2)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/leopold/Applications/miniconda3/envs/aiida_zeopp/lib/python2.7/site-packages/aiida/utils/fixtures.py", line 462, in setUpClass
    cls.fixture_manager.create_profile()
  File "/Users/leopold/Applications/miniconda3/envs/aiida_zeopp/lib/python2.7/site-packages/aiida/utils/fixtures.py", line 186, in create_profile
    'AiiDA dbenv can not be loaded while creating a test profile')
FixtureError: 'AiiDA dbenv can not be loaded while creating a test profile'

----------------------------------------------------------------------
Ran 1 test in 8.051s

FAILED (errors=1)
```

**Comment by ltalirz** [comment]
>  the unit test docs are not very explicit on that.

I find [this](https://docs.python.org/2/library/unittest.html#class-and-module-fixtures) section quite helpful.
> I guess this could be solved by keeping a count on Plugin class instances and only setting up the test environment the first time.

It is not difficult to check in `setupClass` whether the db environment is already set up.
What I haven't been able to figure out, though: how to destroy the environment again after the last test has run.

I've now looked into this a bit and if someone simply runs `python -m unittest discover`, I don't see a way to destroy the environment after the last test.

This leaves two options:
 1. Properly implement db setup/destroy in `setupClass` and `tearDownClass` so that they can be called multiple times in succession (or better perhaps: `setUpModule` and `tearDownModule` to save time if there are multiple classes in one file) . This requires either an `unload_dbenv` (as needed also for #1416), or one needs to separate out the loading of the dbenv from the rest of the setup (the dbenv will be properly destroyed at the end anyhow)

2. Forbid people to use `python -m unittest discover` and force them to use a custom testrunner script, which sets up and destroys the environment only once (see example in #1451).

I suggest to make 1. the default implementation of the fixtures in aiida. People will be able to use the familiar `python -m unittest discover`.
Some people will then still want to modify this and use 2. instead, since this saves unnecessary can let the tests run significantly faster.

@giovannipizzi What's your take on this?

@DropD Is this an issue in pytest as well or not? If this issue does not exist with pytest, I believe we should recommend to use pytest (and I will modify the plugin template accordingly).

**Comment by ltalirz** [comment]
There is another issue with unittest at the moment - namely, the postgres servers are only destroyed when unittest exits gracefully.
When a test fails, the (temporary) postgres server continues running indefinitely - you can easily end up having tens of servers running.

**Comment by ltalirz** [comment]
Closed via #1622
